SPECS/libhugetlbfs/libhugetlbfs.spec

```
@@ -0,0 +1,379 @@
+Patch9: 0009-Disable-hugepage-backed-malloc-if-__morecore-is-not-.patch
+
+%description
+libhugetlbfs is a library which provides easy access to huge pages of memory.
+It is a wrapper for the hugetlbfs file system. Applications can use huge pages
+to fulfill malloc() requests without being recompiled by using LD_PRELOAD.
+Alternatively, applications can be linked against libhugetlbfs without source
+modifications to load BSS or BSS, data, and text segments into large pages.
+
+%package devel
+Summary:  Header files for libhugetlbfs
+Group:    Development/Libraries
+Requires: %{name} = %{version}-%{release}
+
+%description devel
+Contains header files for building with libhugetlbfs.
+
+%package utils
+Summary:  Userspace utilities for configuring the hugepage environment
+Group:    Applications/System
+Requires: %{name} = %{version}-%{release}
+Requires: perl
+Requires: python3
+
+%description utils
+This packages contains a number of utilities that will help administrate the
+use of huge pages on your system.  hugeedit modifies binaries to set default
+segment remapping behavior. hugectl sets environment variables for using huge
+pages and then execs the target program. hugeadm gives easy access to huge page
+pool size control. pagesize lists page sizes available on the machine.
+
+%package tests
+Summary:  Test cases to help on validating the library environment
+Group:    Development/Libraries
+Requires: %{name}-utils = %{version}-%{release}
+
+%description tests
+This packages contains a number of testcases that will help developers
+to verify the libhugetlbfs functionality and validate the library.
+
+%prep
+%setup -q -n %{name}-%{version}
+# downstream patches
+%patch0 -p1
+%patch1 -p1
+%patch2 -p1
+%patch3 -p1
+%patch4 -p1
+%patch5 -p1
+%patch6 -p1
+%patch7 -p1
+%patch8 -p1
+%patch9 -p1
+
+sed -i 's/\/usr\/bin\/python/\/usr\/bin\/python3/' huge_page_setup_helper.py
+sed -i 's/\/usr\/bin\/python/\/usr\/bin\/python3/' tests/run_tests.py
+
+%build
+%{set_build_flags}
+# Parallel builds are not reliable
+make all BUILDTYPE=NATIVEONLY V=1
+
+%install
+make install PREFIX=%{_prefix} DESTDIR=${buildroot} LDSCRIPTDIR=%{ldscriptdir} BUILDTYPE=NATIVEONLY
+make install-helper PREFIX=%{_prefix} DESTDIR=${buildroot} LDSCRIPTDIR=%{ldscriptdir} BUILDTYPE=NATIVEONLY
+make install-tests PREFIX=%{_prefix} DESTDIR=${buildroot} LDSCRIPTDIR=%{ldscriptdir} BUILDTYPE=NATIVEONLY
+mkdir -p -m755 ${buildroot}%{_sysconfdir}/security/limits.d
+touch ${buildroot}%{_sysconfdir}/security/limits.d/hugepages.conf
+
+# clear execstack flag
+execstack --clear-execstack %{buildroot}/%{_lib64dir}/libhugetlbfs.so
+execstack --clear-execstack %{buildroot}/%{_lib64dir}/libhugetlbfs_privutils.so
+
+# remove statically built libraries:
+rm -f %{buildroot}/%{_lib64dir}/*.a
+rm -f %{buildroot}/%{_lib64dir}/libhugetlbfs/tests/*/*.link*
+# remove unused sbin directory
+rm -fr %{buildroot}/%{_sbindir}/
+
+%post -p /sbin/ldconfig
+%postun -p /sbin/ldconfig
```

Nit: `%ldconfig_scriptlets` is the same as adding both of these lines

```suggestion
%ldconfig_scriptlets
```